Zend/zend_builtin_functions.c

```diff
@@ -1273,9 +1270,19 @@ ZEND_FUNCTION(get_object_vars)
 						const char *prop_name, *class_name;
 						size_t prop_len;
 						zend_unmangle_property_name_ex(key, &class_name, &prop_name, &prop_len);
+						/* We assume here that a mangled property name is never
+						 * numeric. This is probably a safe assumption, but
+						 * theoretically someone might write an extension with
+						 * private, numeric properties. Well, too bad.
```

Comment:
In theory, yes, but bear in mind you can't just stick a null byte in front of anything here, you'd have to deliberately form a correct mangled property name that `zend_unmangle_property_name_ex` won't barf at here.